subxt/src/client/light_client/mod.rs

```diff
@@ -115,11 +137,36 @@ impl<T: Config> LightClient<T> {
     pub fn runtime_api(&self) -> RuntimeApiClient<T, Self> {
         <Self as OfflineClientT<T>>::runtime_api(self)
     }
+
+    /// Returns the chain ID of the current light-client.
+    pub fn chain_id(&self) -> subxt_lightclient::ChainId {
+        self.raw_rpc.chain_id()
+    }
+
+    /// Target a different chain identified by the provided chain ID for requests.
+    ///
+    /// The provided chain ID is provided by the `smoldot_light::Client::add_chain` and it must
+    /// match one of the `smoldot_light::JsonRpcResponses` provided in [`Self::new_from_client`].
+    ///
+    /// # Note
+    ///
+    /// This uses the same underlying instance created by [`Self::new_from_client`].
+    pub async fn target_chain<TChainConfig: Config>(
+        &self,
+        chain_id: subxt_lightclient::ChainId,
+    ) -> Result<LightClient<TChainConfig>, crate::Error> {
+        let raw_rpc = self.raw_rpc.target_chain(chain_id);
+        let rpc_client = RpcClient::new(raw_rpc.clone());
+
+        let client = OnlineClient::<TChainConfig>::from_rpc_client(rpc_client).await?;
+
+        Ok(LightClient { client, raw_rpc })
+    }
```

Comment:
There's something I find a little odd about this being on the `LightClient` itself (and also I think it's the reason the `LightClient` needs to carry around the `raw_rpc` thing).

I wonder whether it'd make sense to instead have something like:

```rust
// Instead of `build` spitting out a subxt client immediately, it spits out a RawLightClient:
let raw_light_client: RawLightClient = RawLightClientBuilder::new().add_chain(..).add_chain(..).build(client);

// Turn `RawLightClient` into a subxt client as many times as we want like this:
let light_client: LightClient<PolkadotConfig> = raw_light_client.for_chain(chain_id);
let light_client2: LightClient<SubstrateConfig> = raw_light_client.for_chain(other_chain_id);
```